src/App/Fossa/Analyze/ScanSummary.hs

```diff
@@ -0,0 +1,244 @@
+summarizeProjectScan (Scanned dpi (Failure _ _)) = failColorCoded $ renderDiscoveredProjectIdentifier dpi <> renderFailed
+summarizeProjectScan (Scanned _ (Success wg pr)) = successColorCoded wg $ renderProjectResult pr <> renderSucceeded wg
+summarizeProjectScan (SkippedDueToProvidedFilter dpi) = renderDiscoveredProjectIdentifier dpi <> renderSkipped
+summarizeProjectScan (SkippedDueToDefaultProductionFilter dpi) = renderDiscoveredProjectIdentifier dpi <> renderSkipped
+
+---------- Rendering Helpers
+
+logInfoVsep :: (Has Logger sig m) => [Doc AnsiStyle] -> m ()
+logInfoVsep = traverse_ logInfo
+
+renderDiscoveredProjectIdentifier :: DiscoveredProjectIdentifier -> Doc AnsiStyle
+renderDiscoveredProjectIdentifier dpi = renderProjectPathAndType (dpiProjectType dpi) (dpiProjectPath dpi)
+
+renderProjectResult :: ProjectResult -> Doc AnsiStyle
+renderProjectResult pr = renderProjectPathAndType (projectResultType pr) (projectResultPath pr)
+
+renderProjectPathAndType :: DiscoveredProjectType -> Path Abs Dir -> Doc AnsiStyle
+renderProjectPathAndType pt path =
+  (annotate bold . pretty $ projectTypeToText pt)
+    <> pretty (" project in " :: Text)
+    <> pretty (toText path)
+
+successColorCoded :: [EmittedWarn] -> Doc AnsiStyle -> Doc AnsiStyle
+successColorCoded ew = if countWarnings ew == 0 then annotate (color Green) else annotate (color Yellow)
+
+failColorCoded :: Doc AnsiStyle -> Doc AnsiStyle
+failColorCoded = annotate (color Red)
+
+renderSkipped :: Doc ann
+renderSkipped = pretty (": skipped" :: Text)
+
+renderSucceeded :: [EmittedWarn] -> Doc ann
+renderSucceeded ew =
+  if countWarnings ew == 0
+    then pretty (": succeeded" :: Text)
+    else pretty (": succeeded with " :: Text) <> pretty (show $ countWarnings ew) <> pretty (": warning" :: Text)
+
+renderFailed :: Doc ann
+renderFailed = pretty (": failed" :: Text)
+
+orderByScanStatusAndType :: DiscoveredProjectScan -> DiscoveredProjectScan -> Ordering
+orderByScanStatusAndType (SkippedDueToProvidedFilter lhs) (SkippedDueToProvidedFilter rhs) = compare lhs rhs
+orderByScanStatusAndType (SkippedDueToProvidedFilter lhs) (SkippedDueToDefaultProductionFilter rhs) = compare lhs rhs
+orderByScanStatusAndType (SkippedDueToDefaultProductionFilter lhs) (SkippedDueToProvidedFilter rhs) = compare lhs rhs
+orderByScanStatusAndType (SkippedDueToDefaultProductionFilter lhs) (SkippedDueToDefaultProductionFilter rhs) = compare lhs rhs
+orderByScanStatusAndType (SkippedDueToDefaultProductionFilter _) (Scanned _ _) = GT
+orderByScanStatusAndType (SkippedDueToProvidedFilter _) (Scanned _ _) = GT
+orderByScanStatusAndType (Scanned _ (Success _ lhs)) (Scanned _ (Success _ rhs)) = compare (projectResultType lhs) (projectResultType rhs)
+orderByScanStatusAndType (Scanned lhs (Failure _ _)) (Scanned rhs (Failure _ _)) = compare lhs rhs
+orderByScanStatusAndType (Scanned _ (Success _ _)) (Scanned _ (Failure _ _)) = LT
+orderByScanStatusAndType (Scanned _ _) _ = LT
```

Comment:
`DiscoveredProjectScan` should implement `Ord` using this function, rather than just defining it freely.

Reply:
Fixed in  https://github.com/fossas/fossa-cli/pull/790/commits/16c80c4997a2c3812aa1c781bb9bf657541b9b7c